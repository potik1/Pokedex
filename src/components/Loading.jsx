import React from 'react';

const Loading = () => (
  <div className="col-8  text-center">
    <div className="loader"/>
  </div>
);

export default Loading;
