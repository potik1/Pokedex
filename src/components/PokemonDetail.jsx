import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { renderLoading } from '../utils/functions';
import ShowingArrow from './Showing';


class PokemonDetail extends Component {
  componentDidUpdate() {
    this.scroll();
  }

  scroll = () => {
    if (this.element) {
      this.element.scrollIntoView({ behavior: 'smooth' });
    }
  };

  renderPokemonDetails = (pokemon) => {
    const { name, weight, height } = pokemon;
    const index = pokemon.id;
    const imgFrontShiny = pokemon.sprites.front_shiny;
    const imgBackShiny = pokemon.sprites.back_shiny;
    const baseExp = pokemon.base_experience;

    const tableStat = pokemon.stats.map(stat => (
      <tr key={stat.stat.name}>
        <td>{stat.stat.name}</td>
        <td>{stat.base_stat}</td>
      </tr>
    ));

    const typeStat = pokemon.types.map(type => (
      <li key={type.type.name}>{type.type.name}</li>
    ));
    return (
      <div>
        <div className="row my-2">
          <div className="col-12 text-uppercase">
            <h4 className="text-center font-weight-bold">{name}</h4>
          </div>
        </div>
        <div className="row my-2">
          <div className="col-12 text-center">
            <img src={imgFrontShiny} alt="imgFrontShiny" />
            <img src={imgBackShiny} alt="imgBackShiny" />
          </div>
        </div>
        <div className="row">
          <div className="col-6  mx-auto">
            <h5 className="font-italic font-weight-bold">
              Characteristics:
            </h5>
            <ul>
              <li>Index: {index}</li>
              <li>Height: {height}</li>
              <li>Weight: {weight}</li>
              <li>Basic experience: {baseExp}</li>
            </ul>
          </div>
          <div className="col-6  mx-auto">
            <h5 className="font-italic font-weight-bold">Types:</h5>
            <ul>
              {typeStat}
            </ul>
          </div>
        </div>
        <div className="row my-4 mx-2">
          <h5 className="font-italic font-weight-bold">
            Statistics:
          </h5>
          <table className="table table-sm table-bordered">
            <thead>
              <tr className="table-secondary">
                <th>Ability</th>
                <th>Value</th>
              </tr>
            </thead>
            <tbody>
              {tableStat}
            </tbody>
          </table>
        </div>
      </div>
    );
  };

  renderPokemonNotFound = () => (
    <div className="row">
      <h2 className="text-danger mx-auto">Pokemon Not Found!</h2>
    </div>
  );

  render() {
    return (
      <div ref={(el) => { this.element = el; }}>
        {this.props.loading && renderLoading()}
        {this.props.pokemonNotFound && this.renderPokemonNotFound()}
        {this.props.pokemon
          && !this.props.loading
          && !this.props.pokemonNotFound
          && this.renderPokemonDetails(this.props.pokemon)}
        <ShowingArrow />
      </div>
    );
  }
}

const mapStateToProps = state => (
  {
    pokemon: state.pokemons.pokemon,
    pokemonNotFound: state.pokemons.pokemonNotFound,
    loading: state.pokemons.loading,
  }
);

PokemonDetail.propTypes = {
  pokemonNotFound: PropTypes.bool.isRequired,
  /* eslint-disable */
  pokemon: PropTypes.object,
  /* eslint-enable */
  loading: PropTypes.bool.isRequired,
};

export default connect(mapStateToProps, null)(PokemonDetail);

