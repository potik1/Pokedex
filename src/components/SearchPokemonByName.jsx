import React, {Component} from 'react';
import {connect} from 'react-redux';
import {searchPokemon } from '../actions/index';

class SearchPokemonByName extends Component {

  constructor(props) {
    super(props);
    this.state = {term: ''};
  }

  onInputChange(event) {
    this.setState({term: event.target.value});
  }

  onFormSubmit(event) {
    event.preventDefault();
    this.props.searchPokemon(this.state.term);
    this.setState({term:''});
  }

  render() {
    return (
      <form
        onSubmit={(event)=> this.onFormSubmit(event)}
        className="form-inline my-2 my-lg-0 ">
        <input
          placeholder="Enter pokemon name..."
          className="form-control"
          value={this.state.term}
          onChange={(event) => this.onInputChange(event)}
        />
        <span className="input-group-btn ">
               <button type="submit"
                       className="btn btn-secondary">Submit</button>
          </span>
      </form>
    );
  }
}

export default connect(null, {searchPokemon})(SearchPokemonByName);
