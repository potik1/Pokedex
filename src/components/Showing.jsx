import React from 'react';
import _ from 'lodash';

const Showing = () => {
  const arrowDiv = document.getElementById('arrow');

  window.addEventListener('scroll', _.debounce(() => {
    const scroll = {
      x: window.pageXOffset,
      y: window.pageYOffset,
    };

    if (scroll.y <= 300 && arrowDiv !== null) {
      arrowDiv.classList.add('d-none');
    } else if (scroll.y > 300 && arrowDiv !== null) {
      arrowDiv.classList.remove('d-none');
    }
  }, 200));

  return (
    <div>
      <div
        id="arrow"
        role="button"
        className="d-block d-sm-none box"
        onClick={() => window.scroll(0, 0)}
      >
        <div
          className="arrow-up d-sm-none"
        />
      </div>
    </div>
  );
};

export default Showing;
