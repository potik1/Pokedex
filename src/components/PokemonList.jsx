import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { getPokemonList, getPokemon, searchPokemon } from '../actions/index';
import PokemonDetail from './PokemonDetail';

class PokemonList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      page: 1,
    };

    this.props.getPokemonList(this.state.page).then((res) => {
      const pokemonName = res.payload.data.results[0].name;
      this.props.searchPokemon(pokemonName);
    });

    this.activeEl = null;
  }

  getSelectPokemonList = pokemonUrl => (e) => {
    this.props.getPokemon(pokemonUrl);
    if ((this.activeEl) && (e.target !== this.activeEl)) {
      this.activeEl.classList.remove('font-weight-bold');
    }
    e.target.classList.add('font-weight-bold');
    this.activeEl = e.target;
  };

  nextPage() {
    const page = this.state.page + 1;
    this.setState({ page });
    this.props.getPokemonList(page);
  }

  prevPage() {
    const page = this.state.page - 1;
    this.setState({ page });
    this.props.getPokemonList(page);
  }

  renderList() {
    return (
      this.props.pokemons.map(pokemon =>
        (<li
          className="list-group-item text-capitalize"
          key={pokemon.name}
          onClick={this.getSelectPokemonList(pokemon.url)}
        >
          {pokemon.name}
        </li>))
    );
  }

  render() {
    return (
      <div className="container">
        <div className="row">
          <h3 className="mt-4 mx-auto"> LIST OF POKEMONS</h3>
          <div className="col-12 " />
          <div className="col-lg-3 col-md-3 col-sm-5 col-xs-6 mx-auto">
            <div className="text-center">
              <button
                className={this.props.previousPage
                  ? 'btn btn-outline-info'
                  : 'btn btn-outline-info disabled'}
                onClick={() => this.prevPage()}
              >
                Previous
              </button>
              <button
                className={this.props.nextPage
                  ? 'btn btn-outline-info'
                  : 'btn btn-outline-info disabled'}
                onClick={() => this.nextPage()}
              >
                Next
              </button>
            </div>
            <div >
              <ul className="list-group my-4">
                {this.renderList()}
              </ul>
            </div>
            <hr className="d-block d-sm-none" />
          </div>
          <div className="col-lg-8 col-md-8 col-sm-7 col-xs-6 mx-auto" >
            <PokemonDetail />
          </div>
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    pokemons: state.pokemons.pokemons,
    previousPage: state.pokemons.previousPage,
    nextPage: state.pokemons.nextPage,
    pokemonNotFound: state.pokemons.pokemonNotFound,
  };
}

PokemonList.propTypes = {
  getPokemonList: PropTypes.func.isRequired,
  getPokemon: PropTypes.func.isRequired,
  searchPokemon: PropTypes.func.isRequired,
  /* eslint-disable */
  pokemons: PropTypes.arrayOf(PropTypes.object).isRequired,
  /* eslint-disable */
  previousPage: PropTypes.string.isRequired,
  /* eslint-disable */
  nextPage: PropTypes.string.isRequired,
};

export default connect(
  mapStateToProps,
  { getPokemon, getPokemonList, searchPokemon },
)(PokemonList);

