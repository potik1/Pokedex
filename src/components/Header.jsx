import React from 'react';
import SearchPokemonByName from './SearchPokemonByName';

const Header = () => (
  <nav className="navbar bg-secondary text-white">
    <a className="navbar-brand">POKEDEX</a>
      <SearchPokemonByName  />
  </nav>
);

export default Header;
