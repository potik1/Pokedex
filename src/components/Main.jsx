import React, { Component } from 'react';
import Header from './Header';
import PokemonList from './PokemonList';
import PokemonDetail from './PokemonDetail';

class Main extends Component {
  render() {
    return (
      <div>
        <Header />
        <PokemonList />
      </div>
    );
  }
}

export default Main;
