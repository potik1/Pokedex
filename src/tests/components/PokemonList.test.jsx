import React from 'react';
import { shallow } from 'enzyme';
import { PokemonList } from '../../components/PokemonList';

describe('Pokemon List', () => {
  const mockgetPokemonList = jest.fn();
  const mockgetPokemon = jest.fn();
  const mockhasError = jest.fn();
  const mockerrorLoading = jest.fn();

  let props = {
    page: 1,
    activeEl: null,
    loadingList: false,
    pokemonNotFound: false,
    hasError: false,
    nextPage: '',
    previousPage: '',
    pokemons: [],
    getPokemonList: mockgetPokemonList,
    getPokemon: mockgetPokemon,
  };

  let pokemonList = shallow(<PokemonList {...props} />);

  test('renders correctly', () => {
    expect(pokemonList).toMatchSnapshot();
  });
  test('state at the start', () => {
    expect(pokemonList.state().page).toEqual(1);
    expect(pokemonList.state().activeEl).toEqual(undefined);
  });

  test('contains Connected PokemonDetail component', () => {
    expect(pokemonList.find('Connect(PokemonDetail)').exists()).toBe(true);
  });

  describe(
    'calls `getPokemonList()` and `getPokemon()` when componentDidMount',
    () => {
      beforeEach(() => {
        pokemonList.instance().componentDidMount();
        props = {url1: 'https://pokeapi.co/api/v2/pokemon/bulbasaur'};
      });
      test(
        'calls `getPokemonList()` and `getPokemon()`',
        () => {
          expect(mockgetPokemonList).toHaveBeenCalledTimes(1);
          expect(mockgetPokemon).toBeCalledWith(props.url1);
        },
      );
    },
  );

  describe('renderList', () => {
    beforeEach(() => {
      props = {
        pokemons: [{name: 'bulbasaur'}, {name: 'ivysaur'}],
        page: 1,
        activeEl: null,
        loadingList: false,
        pokemonNotFound: false,
        hasError: false,
        nextPage: '',
        previousPage: '',
        getPokemonList: mockgetPokemonList,
        getPokemon: mockgetPokemon,
        url1: 'https://pokeapi.co/api/v2/pokemon/bulbasaur',
      };

      pokemonList = shallow(<PokemonList {...props} />);
    });

    test('renders correctly', () => {
      expect(pokemonList.find('ul')).toHaveLength(1);
      expect(pokemonList.find('li')).toHaveLength(2);
      expect(pokemonList.find('ul').childAt(0).text()).toEqual('bulbasaur');
      expect(pokemonList.find('ul').childAt(1).text()).toEqual('ivysaur');
    });

    describe('select pokemon when clicking', () => {
      test('before clicking has a class `list-group-item` and has not `font-weight-bold`', () => {
        expect(pokemonList.find('ul').childAt(0).hasClass('list-group-item')).toEqual(true);
        expect(pokemonList.find('ul').childAt(0).hasClass('font-weight-bold')).toEqual(false);
      });
      test('add classList `font-weight-bold` to selected pokemon', () => {
        const event = { target: { name: 'bulbasaur' } };
        const getSelectPokemonList = jest.fn();
        pokemonList.find('ul').childAt(0).simulate('click', event);
        getSelectPokemonList(props.url1);
        //pokemonList.update();
        //expect(targetOn).toContain('.font-weight-bold');
      });
    });
  });
});
