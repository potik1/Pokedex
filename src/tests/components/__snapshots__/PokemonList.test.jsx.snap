// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon List renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PokemonList
    activeEl={null}
    getPokemon={[Function]}
    getPokemonList={[Function]}
    hasError={false}
    loadingList={false}
    nextPage=""
    page={1}
    pokemonNotFound={false}
    pokemons={Array []}
    previousPage=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
>
        <h3
                className="mt-4 mx-auto"
        >
                 LIST OF POKEMONS
        </h3>
        <div
                className="col-12 "
        />
        <div
                className="col-lg-3 col-md-3 col-sm-5 col-xs-6 mx-auto"
        >
                <div
                        className="text-center"
                >
                        <button
                                className="btn btn-outline-info disabled"
                                onClick={[Function]}
                        >
                                Previous
                        </button>
                        <button
                                className="btn btn-outline-info disabled"
                                onClick={[Function]}
                        >
                                Next
                        </button>
                </div>
                <div>
                        <ul
                                className="list-group my-4"
                        >
                                
                        </ul>
                </div>
                <hr
                        className="d-block d-sm-none"
                />
        </div>
        <div
                className="col-lg-8 col-md-8 col-sm-7 col-xs-6 mx-auto"
        >
                <Connect(PokemonDetail) />
        </div>
</div>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="mt-4 mx-auto"
>
             LIST OF POKEMONS
</h3>,
          <div
            className="col-12 "
/>,
          <div
            className="col-lg-3 col-md-3 col-sm-5 col-xs-6 mx-auto"
>
            <div
                        className="text-center"
            >
                        <button
                                    className="btn btn-outline-info disabled"
                                    onClick={[Function]}
                        >
                                    Previous
                        </button>
                        <button
                                    className="btn btn-outline-info disabled"
                                    onClick={[Function]}
                        >
                                    Next
                        </button>
            </div>
            <div>
                        <ul
                                    className="list-group my-4"
                        >
                                    
                        </ul>
            </div>
            <hr
                        className="d-block d-sm-none"
            />
</div>,
          <div
            className="col-lg-8 col-md-8 col-sm-7 col-xs-6 mx-auto"
>
            <Connect(PokemonDetail) />
</div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " LIST OF POKEMONS",
            "className": "mt-4 mx-auto",
          },
          "ref": null,
          "rendered": " LIST OF POKEMONS",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "col-12 ",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="text-center"
>
                <button
                                className="btn btn-outline-info disabled"
                                onClick={[Function]}
                >
                                Previous
                </button>
                <button
                                className="btn btn-outline-info disabled"
                                onClick={[Function]}
                >
                                Next
                </button>
</div>,
              <div>
                <ul
                                className="list-group my-4"
                >
                                
                </ul>
</div>,
              <hr
                className="d-block d-sm-none"
/>,
            ],
            "className": "col-lg-3 col-md-3 col-sm-5 col-xs-6 mx-auto",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="btn btn-outline-info disabled"
                    onClick={[Function]}
>
                    Previous
</button>,
                  <button
                    className="btn btn-outline-info disabled"
                    onClick={[Function]}
>
                    Next
</button>,
                ],
                "className": "text-center",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Previous",
                    "className": "btn btn-outline-info disabled",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Previous",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Next",
                    "className": "btn btn-outline-info disabled",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Next",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ul
                  className="list-group my-4"
>
                  
</ul>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    false,
                    false,
                    Array [],
                  ],
                  "className": "list-group my-4",
                },
                "ref": null,
                "rendered": Array [
                  false,
                  false,
                ],
                "type": "ul",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "d-block d-sm-none",
              },
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(PokemonDetail) />,
            "className": "col-lg-8 col-md-8 col-sm-7 col-xs-6 mx-auto",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
>
          <h3
                    className="mt-4 mx-auto"
          >
                     LIST OF POKEMONS
          </h3>
          <div
                    className="col-12 "
          />
          <div
                    className="col-lg-3 col-md-3 col-sm-5 col-xs-6 mx-auto"
          >
                    <div
                              className="text-center"
                    >
                              <button
                                        className="btn btn-outline-info disabled"
                                        onClick={[Function]}
                              >
                                        Previous
                              </button>
                              <button
                                        className="btn btn-outline-info disabled"
                                        onClick={[Function]}
                              >
                                        Next
                              </button>
                    </div>
                    <div>
                              <ul
                                        className="list-group my-4"
                              >
                                        
                              </ul>
                    </div>
                    <hr
                              className="d-block d-sm-none"
                    />
          </div>
          <div
                    className="col-lg-8 col-md-8 col-sm-7 col-xs-6 mx-auto"
          >
                    <Connect(PokemonDetail) />
          </div>
</div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="mt-4 mx-auto"
>
               LIST OF POKEMONS
</h3>,
            <div
              className="col-12 "
/>,
            <div
              className="col-lg-3 col-md-3 col-sm-5 col-xs-6 mx-auto"
>
              <div
                            className="text-center"
              >
                            <button
                                          className="btn btn-outline-info disabled"
                                          onClick={[Function]}
                            >
                                          Previous
                            </button>
                            <button
                                          className="btn btn-outline-info disabled"
                                          onClick={[Function]}
                            >
                                          Next
                            </button>
              </div>
              <div>
                            <ul
                                          className="list-group my-4"
                            >
                                          
                            </ul>
              </div>
              <hr
                            className="d-block d-sm-none"
              />
</div>,
            <div
              className="col-lg-8 col-md-8 col-sm-7 col-xs-6 mx-auto"
>
              <Connect(PokemonDetail) />
</div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " LIST OF POKEMONS",
              "className": "mt-4 mx-auto",
            },
            "ref": null,
            "rendered": " LIST OF POKEMONS",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "col-12 ",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="text-center"
>
                  <button
                                    className="btn btn-outline-info disabled"
                                    onClick={[Function]}
                  >
                                    Previous
                  </button>
                  <button
                                    className="btn btn-outline-info disabled"
                                    onClick={[Function]}
                  >
                                    Next
                  </button>
</div>,
                <div>
                  <ul
                                    className="list-group my-4"
                  >
                                    
                  </ul>
</div>,
                <hr
                  className="d-block d-sm-none"
/>,
              ],
              "className": "col-lg-3 col-md-3 col-sm-5 col-xs-6 mx-auto",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      className="btn btn-outline-info disabled"
                      onClick={[Function]}
>
                      Previous
</button>,
                    <button
                      className="btn btn-outline-info disabled"
                      onClick={[Function]}
>
                      Next
</button>,
                  ],
                  "className": "text-center",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Previous",
                      "className": "btn btn-outline-info disabled",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Previous",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Next",
                      "className": "btn btn-outline-info disabled",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Next",
                    "type": "button",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <ul
                    className="list-group my-4"
>
                    
</ul>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      false,
                      false,
                      Array [],
                    ],
                    "className": "list-group my-4",
                  },
                  "ref": null,
                  "rendered": Array [
                    false,
                    false,
                  ],
                  "type": "ul",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "d-block d-sm-none",
                },
                "ref": null,
                "rendered": null,
                "type": "hr",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(PokemonDetail) />,
              "className": "col-lg-8 col-md-8 col-sm-7 col-xs-6 mx-auto",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
